src/checkpoint/container/view_serialize.h

```diff
@@ -93,6 +99,18 @@
 
 namespace checkpoint {
 
+namespace {
+
+template <typename T, typename U>
+void deepCopyWithLocalFence(T& dst, U& src) {
+  // Create and use an execution space to avoid a global Kokkos::fence()
+  auto exec_space = Kokkos::HostSpace::execution_space{};
+  Kokkos::deep_copy(exec_space, dst, src);
+  exec_space.fence();
```

Comment:
This may be effectively `Kokkos::fence()`, depending on implementation details. I'll try to see if there's an alternative